### ISSUE TITLE
No graphics output - "Guest has not initialized the display (yet)"

### ISSUE THREAD

**Issue by Spargeltarzan** [01org/gvt-linux#45]
I use qemu 2.12 and libvirt 4.3 and I struggle with booting my Windows machine with the mdev Intel vgpu attached.
My issue is also "Guest has not initialized the display (yet)" like from DocMAX in issue 20, but the proposed solution to attach a qxl graphics card leads to black screen for me. I tried to connect with virt-manager and virt-viewer, both without success.

I can only boot with graphical output when I remove the mdev vgpu again, but than Windows does not let me install the Intel graphics driver: "The system does not fulfil the minimum requirements" - obviously because no Intel vgpu is attached.

Help is very much appreciated!

**Comment by TerrenceXu** [comment]
If you want to show graphic in libvirt with MDEV, you must boot up with with dma_buf, the wiki as below:
https://github.com/intel/gvt-linux/wiki/Dma_Buf_User_Guide

**Comment by l-cd** [comment]
> I use qemu 2.12 and libvirt 4.3 and I struggle with booting my Windows machine with the mdev Intel vgpu attached.
> My issue is also "Guest has not initialized the display (yet)" like from DocMAX in issue 20, but the proposed solution to attach a qxl graphics card leads to black screen for me. I tried to connect with virt-manager and virt-viewer, both without success.
> 
> I can only boot with graphical output when I remove the mdev vgpu again, but than Windows does not let me install the Intel graphics driver: "The system does not fulfil the minimum requirements" - obviously because no Intel vgpu is attached.
> 
> Help is very much appreciated!

Has the issue been solved? How to solve it?

**Comment by rugubara** [comment]
I did follow the dma_buf guide. I have another VM with linux and Virtio driver with 3D  running fine so I assume dma_buf is ok. 
 I have installed the intel 26.20.100.6709 in the Windows 10 guest. The card was detected and running w/o errors in the Device Manager. When I disabled the QXL by setting type='none' I experience  Guest has not initialized the display (yet) message right from the start (i.e. there is no OVMF TianoCore logo during bootup).

I run gentoo kernel 5.1.1, qemu 4.0.0 and libvirt 5.2.0-r1 
anton@PF16W6Y2 ~ $ uname -a
Linux PF16W6Y2 5.1.1-gentoo #1 SMP PREEMPT Sun May 12 23:08:59 MSK 2019 x86_64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz GenuineIntel GNU/Linux
a
